pom.xml

```diff
@@ -659,16 +580,116 @@
 						<artifactId>nexus-staging-maven-plugin</artifactId>
-						<version>${nexus-staging-plugin.version}</version>
 						<extensions>true</extensions>
 						<configuration>
 							<serverId>ossrh</serverId>
 							<nexusUrl>https://oss.sonatype.org/</nexusUrl>
 							<autoReleaseAfterClose>false</autoReleaseAfterClose>
 						</configuration>
 					</plugin>
+					<plugin>
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-javadoc-plugin</artifactId>
+						<executions>
+							<execution>
+								<id>javadoc</id>
+								<goals>
+									<goal>jar</goal>
+								</goals>
+								<phase>package</phase>
+							</execution>
+							<execution>
+								<id>aggregate</id>
+								<goals>
+									<goal>aggregate</goal>
+								</goals>
+							</execution>
+						</executions>
+						<configuration>
+							<!-- https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8212233 -->
+							<source>8</source>
+							<excludePackageNames>
+								com.example,
+								com.example.*,
+								<!-- Exclude samples and classes in the Firestore Google Client library packages -->
+								com.google.cloud.firestore
+							</excludePackageNames>
+							<links>
+								<link>https://googleapis.dev/java/google-cloud-vision/latest/</link>
+								<link>https://googleapis.dev/java/google-cloud-spanner/latest/</link>
+								<link>https://googleapis.dev/java/google-cloud-clients/latest/</link>
+							</links>
+						</configuration>
+					</plugin>
+					<plugin>
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-source-plugin</artifactId>
+						<executions>
+							<execution>
+								<id>attach-sources</id>
+								<goals>
+									<goal>jar</goal>
+								</goals>
+								<phase>package</phase>
+							</execution>
+						</executions>
+					</plugin>
+					<plugin>
+						<groupId>org.codehaus.mojo</groupId>
+						<artifactId>flatten-maven-plugin</artifactId>
+						<inherited>true</inherited>
+						<executions>
+							<execution>
+								<!-- Tidy up all POMs before they are published -->
+								<id>flatten</id>
+								<phase>process-resources</phase>
+								<goals>
+									<goal>flatten</goal>
+								</goals>
+								<configuration>
+									<updatePomFile>true</updatePomFile>
+									<flattenMode>oss</flattenMode>
+									<pomElements>
+										<parent>expand</parent>
+										<distributionManagement>remove</distributionManagement>
+										<repositories>remove</repositories>
+									</pomElements>
+								</configuration>
+							</execution>
+							<execution>
+								<id>flatten.clean</id>
+								<phase>clean</phase>
+								<goals>
+									<goal>clean</goal>
+								</goals>
+							</execution>
+						</executions>
+					</plugin>
 				</plugins>
 			</build>
+			<reporting>
+				<plugins>
+					<plugin>
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-javadoc-plugin</artifactId>
```

Comment:
Do we need to repeat it under `reporting`? We don't seem to be doing it for Spanner Hibernate?

Reply:
You're right, it looks like it's not needed. Removing.